server/gw.go

```diff
@@ -17,9 +17,9 @@ type GRPCGateway struct {
 }
 
 // NewGateway creates a new server.GRPCGateway to proxy grpc requests to specified host and port
-func NewGateway(host string, port int) (*GRPCGateway, error) {
+func NewGateway(host string, port int, opts ...runtime.ServeMuxOption) (*GRPCGateway, error) {
```

Comment:
Can we have a new type for `gatewayOptions` and use that to implement this? It would enable us to use options for more stuff than just `ServerMuxOptions`.

Both [http](https://github.com/odpf/salt/blob/d49affffd790f71aea6fde7fec43b1f428797705/server/http.go#L21) and [grpc](https://github.com/odpf/salt/blob/d49affffd790f71aea6fde7fec43b1f428797705/server/grpc.go#L23) are also doing it in similar manner.

Reply:
sure. makes sense to have. will add changes.